### PR TITLE
Use helm to list search results

### PR DESCRIPTION
**Pull request by bomgar**
Provides an alternative to ensime-search (optional)

### CODE REVIEW

**Comment by fommil** [comment]
This is all great stuff but you're going to have to start thinking about testing. Sooner or later were going to remove everything that doesn't have a test. Can you have a think about that please?

**Comment by bomgar** [comment]
I'm happy to provide some tests but I need help to do that. It is hard to use the current tests as a template.

**Comment by fommil** [comment]
I'd recommend using the cucumber test framework, look under `feature` for some examples, failing that use `ert` (see `test/ensime-emacs-tests.el`). We want to get away from using server integration tests for everything so I recommend learning how to use a mocking framework like https://github.com/rejeep/el-mock.el  or (complementary) https://github.com/nicferrier/emacs-fakir

More generally, it would be really fantastic if we were able to rewrite all our tests to the cucumber with mocking style, then only have one or two smoke tests that launch and connect to the real server.

**Comment by fommil** [comment]
I'm happy for you to do that in a follow up PR, keep it coming! :smile:

File: ensime-client.el [hunk 1]
```diff
@@ -995,6 +995,14 @@ copies. All other objects are used unchanged. List must not contain cycles."
      ,max-results
      )))
 
+(defun ensime-rpc-public-symbol-search
```
Comment (fommil) [inline ensime-client.el]: we probably need to do this for all the API calls, abstract the API from the implementation because it **will** change.

**Comment by fommil** [comment]
any chance of a GIF of this working? I'd love to see it! I'd really like to move search over to ido or something lighter weight by default.

**Comment by bomgar** [comment]
I'll have a look at el-mock.

**Comment by bomgar** [comment]
The colors are a little bit off though
ensime_helm_search

**Pull request merged**